pkg/kubectl/cmd/util/factory.go

```diff
@@ -106,8 +108,23 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 		clientConfig = DefaultClientConfig(flags)
 	}
 
-	clients := NewClientCache(clientConfig)
+	clients := newClientCache(clientConfig)
 
+	var experimentalClient *client.ExperimentalClient
+	cfg, experimentalClientErr := clientConfig.ClientConfig()
```

Comment:
Client config can fail - it can fail BEFORE the help message runs.  So you can't allow CLI to have Factory creation fail based on a bad config - it has to fail later.